driver-core/src/main/java/com/datastax/driver/core/ControlConnection.java

```diff
@@ -571,20 +585,26 @@ private static void updateInfo(
     // After CASSANDRA-9603 (2.0.17, 2.1.8, 2.2.0 rc2) local row contains one more column:
     // - listen_address
 
-    InetAddress broadcastAddress = null;
+    InetSocketAddress broadcastAddress = null;
     if (row.getColumnDefinitions().contains("peer")) { // system.peers
-      broadcastAddress = row.getInet("peer");
+      int broadcastPort =
```

Comment:
Instead of zero, shouldn't we use the default port number (9042)?

Reply:
This is for the broadcast/listen address which is typically 7000.  I felt that since we can't know for sure what it's configured to, using 0 would be better.

Reply:
Oh sorry, of course. 👍